src/id.rs

```diff
@@ -96,21 +98,20 @@ impl parsec::SecretId for FullId {
     }
 
     fn sign_detached(&self, data: &[u8]) -> <Self::PublicId as parsec::PublicId>::Signature {
-        self.signing_private_key().sign_detached(data)
+        self.sign(data)
     }
 
-    fn encrypt<M: AsRef<[u8]>>(&self, to: &Self::PublicId, msg: M) -> Option<Vec<u8>> {
-        let shared_secret = self
-            .encrypting_private_key()
-            .shared_secret(to.encrypting_public_key());
-        shared_secret.encrypt_bytes(msg.as_ref()).ok()
+    fn encrypt<M: AsRef<[u8]>>(&self, to: &Self::PublicId, plaintext: M) -> Option<Vec<u8>> {
+        serialise(&to.public_encryption_key().encrypt(plaintext)).ok()
```

Comment:
Is it ok to create a new OsRng every time we encrypt something?
Does that allow reproducible seeds? 
```
    /// Encrypts the message using the OS random number generator.
    ///
    /// Uses the `OsRng` by default. To pass in a custom random number generator, use
    /// `encrypt_with_rng()`.
    pub fn encrypt<M: AsRef<[u8]>>(&self, msg: M) -> Ciphertext {
        self.encrypt_with_rng(&mut OsRng::new().expect(ERR_OS_RNG), msg)
    }
```

Reply:
Good catch! I didn't realize it uses rng under the hood. It makes sense though given this thing doesn't use nonces. I'll have to think about how to handle this.

Reply:
Think we are OK here encrypt/decrypt is tested in the BLS crate itself. It is unlikely to fail in any reproducible manner that would be useful.